src/Components/Components/src/Reflection/PropertyGetter.cs

```diff
@@ -0,0 +1,56 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics.CodeAnalysis;
+using System.Reflection;
+using System.Runtime.CompilerServices;
+
+namespace Microsoft.AspNetCore.Components.Reflection;
+
+internal sealed class PropertyGetter
+{
+    private static readonly MethodInfo CallPropertyGetterOpenGenericMethod =
+        typeof(PropertyGetter).GetMethod(nameof(CallPropertyGetter), BindingFlags.NonPublic | BindingFlags.Static)!;
+
+    private readonly Func<object, object?> _GetterDelegate;
+
+    [UnconditionalSuppressMessage(
+        "ReflectionAnalysis",
+        "IL2060:MakeGenericMethod",
+        Justification = "The referenced methods don't have any DynamicallyAccessedMembers annotations. See https://github.com/mono/linker/issues/1727")]
+    public PropertyGetter(Type targetType, PropertyInfo property)
+    {
+        if (property.GetMethod == null)
+        {
+            throw new InvalidOperationException("Cannot provide a value for property " +
+                $"'{property.Name}' on type '{targetType.FullName}' because the property " +
+                "has no getter.");
+        }
+
+        if (RuntimeFeature.IsDynamicCodeSupported)
+        {
+            var getMethod = property.GetMethod;
+
+            var propertyGetterAsFunc =
+                getMethod.CreateDelegate(typeof(Func<,>).MakeGenericType(targetType, property.PropertyType));
+            var callPropertyGetterClosedGenericMethod =
+                CallPropertyGetterOpenGenericMethod.MakeGenericMethod(targetType, property.PropertyType);
+            _GetterDelegate = (Func<object, object>)
+                callPropertyGetterClosedGenericMethod.CreateDelegate(typeof(Func<object, object>), propertyGetterAsFunc);
+        }
+        else
+        {
+            _GetterDelegate = property.GetValue;
```

Comment:
Is the expectation that this is identical in behavior to the branch above, but slower? I'm not clear on why it would necessarily be faster to use the MakeGenericMethod-based approach above than to use property.GetValue in all cases. Would be interested in any details on that.

Reply:
I copy/pasted this from `PropertySetter` and adjusted the code. I'm not familiar with the details, and I'm simply working on the assumption that we did this this way for a reason inside `PropertySetter`. It's something we can revisit separately if needed.